### PR TITLE
Capture TS migration

### PR DESCRIPTION
**Pull request by nulrich**
# Problem

Some Capture components are still in Javascript

# Solution

Convert to Typescript

## Checklist

_put `n/a` if item is not relevant to PR changes_

- [ ] Has the CHANGELOG been updated?
- [ ] Has the README been updated?
- [ ] Has the CONTRIBUTING doc been updated?
- [ ] Has the RELEASE_GUIDELINES been updated?
- [ ] Has the TESTING_GUIDELINES been updated?
- [ ] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [ ] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [ ] Have any new automated tests been implemented or the existing ones changed?
- [ ] Have any new manual tests been written down or the existing ones changed?
- [ ] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

File: src/components/Capture/Face.tsx [hunk 1]
```diff
@@ -108,19 +125,26 @@ class Face extends Component {
...
+    const withLocalisedProps: WithLocalisedProps = {
+      ...this.props,
+    }
+
+    const withTrackingProps: WithTrackingProps = {
+      ...this.props,
+    }
+
+    const stepComponentFaceProps: StepComponentFaceProps = {
+      ...this.props,
```
Comment (DannyvanderJagt) [inline src/components/Capture/Face.tsx:147]: What's the purpose of those three? I think we can just use `{...this.props}` in the components below

File: src/components/Capture/Face.tsx [hunk 2]
```diff
@@ -132,14 +156,17 @@ class Face extends Component {
...
-      const ariaLabelForSelfieCameraView = translate(
-        'selfie_capture.frame_accessibility'
-      )
       if (requestedVariant === 'video') {
         return (
           <FaceVideo
             {...cameraProps}
+            {...withLocalisedProps}
+            {...withTrackingProps}
+            {...stepComponentFaceProps}
```
Comment (DannyvanderJagt) [inline src/components/Capture/Face.tsx:165]: See comment above

File: src/components/Photo/Selfie.tsx [hunk 3]
```diff
@@ -130,10 +147,20 @@ export default class SelfieCapture extends Component {
...
       isCaptureButtonDisabled, // Capture Button is disabled until camera access is allowed + userMedia stream is ready
     } = this.state
+
+    const cameraProps: Omit<CameraProps, 'buttonType'> = {
+      ...this.props,
+    }
+
+    const withTrackingProps: WithTrackingProps = {
+      ...this.props,
+    }
```
Comment (DannyvanderJagt) [inline src/components/Photo/Selfie.tsx:157]: Same here. I think we can remove those and just use `{...this.props}` below.

File: src/components/Photo/Selfie.tsx [hunk 4]
```diff
@@ -6,32 +6,43 @@ import { ToggleFullScreen } from '../FullScreen'
...
+  renderFallback: RenderFallbackProp
+  inactiveError: ErrorProp
+  useMultipleSelfieCapture: boolean
+  snapshotInterval: number
+} & WithTrackingProps &
+  WithLocalisedProps &
+  WithPageIdProps
+
+export default class SelfieCapture extends Component<Props, State> {
+  webcam?: Webcam
```
Comment (DannyvanderJagt) [inline src/components/Photo/Selfie.tsx:41]: We could use a `createRef` here

**Review by DannyvanderJagt — CHANGES_REQUESTED** [review]
I think I see what you're trying to do here. I would suggest to either remove them and just use `{...this.props}` or separate the props and respect the type like:

```
const withTrackingProps: WithTrackingProps = {
     trackScreen: this.props. trackScreen
}
```

File: src/components/Capture/Face.tsx (hunk 1 above)
Comment (nulrich) [reply src/components/Capture/Face.tsx:147]: I was overwhelmed with props drilling during the migration. I'm putting back `...this.props` 😅

File: src/components/Photo/Selfie.tsx (hunk 4 above)
Comment (nulrich) [reply src/components/Photo/Selfie.tsx]: done

File: src/components/Photo/Selfie.tsx (hunk 3 above)
Comment (nulrich) [reply src/components/Photo/Selfie.tsx]: done

File: src/components/Capture/Face.tsx (hunk 2 above)
Comment (nulrich) [reply src/components/Capture/Face.tsx]: done

**Review by DannyvanderJagt — APPROVED** [review]

**Pull request merged**